crates/env/src/engine/experimental_off_chain/types.rs

```diff
@@ -0,0 +1,68 @@
+// Copyright 2018-2021 Parity Technologies (UK) Ltd.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+//! Contains the necessary conversions from `ink_engine` types to types
+//! of this crate.
+
+use super::{
+    test_api::EmittedEvent,
+    AccountError,
+    Error,
+    OffChainError,
+};
+
+impl From<ink_engine::test_api::EmittedEvent> for EmittedEvent {
+    fn from(evt: ink_engine::test_api::EmittedEvent) -> Self {
+        EmittedEvent {
+            topics: evt.topics,
+            data: evt.data,
+        }
+    }
+}
```

Comment:
`ink_engine` has its own internal type and `ink_env` also needs a type. Since we don't want tight type dependence it's necessary to convert between the two.